server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryRequestStreamObserver.java

```diff
@@ -30,6 +30,8 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(DiscoveryServer.class);
 
   final long streamId;
+  // TODO: Add same fix for listeners
```

Comment:
Can you elaborate here? What is the problem with listeners?

Reply:
Clusters are going through warming and there is same case with listeners. It is possible that RDS response might be sent before LDS what can cause same problem with listeners in warming state. It might happen when there is a lot of changes and many groups waiting for configuration. In test I wrote observer problem and reproduced it for clusters.

Reply:
Ok, can we have an issue for this? I don't want to block this PR since it fixes things, but at the same time, it's good to have visibility that this is a problem.

Reply:
Done: #134